core/src/main/scala/algebra/ring/Ring.scala

```diff
@@ -32,11 +32,11 @@ trait Ring[@sp(Int, Long, Float, Double) A] extends Any with Rig[A] with Rng[A]
   def fromInt(n: Int): A = sumN(one, n)
 }
 
-trait RingFunctions {
+trait RingFunctions extends AdditiveGroupFunctions with MultiplicativeMonoidFunctions {
```

Comment:
Hmm, wondering whether the `fromInt` method should be in `RingFunctions` at all. One can already write

```scala
Ring[A].fromInt(n)
```

with `fromInt` in `RingFunctions` (and thus in `Ring`), one can write

```scala
Ring.fromInt[A](n)
```

I don't see any benefit here.